apps/admin_api/test/admin_api/v1/controllers/blockchain/token_controller_test.exs

```diff
@@ -91,48 +91,148 @@ defmodule AdminAPI.V1.Blockchain.TokenControllerTest do
+          symbol: nil,
+          name: "Bitcoin",
+          amount: 1000,
+          locked: false,
+          description: "desc",
+          subunit_to_unit: 100
+        })
+
+      assert_deploy_erc20(:fails_with_missing_data, response)
+    end
+
+    test_with_auths "fails to deploys an ERC20 token with name = nil", context do
+      enable_blockchain(context)
+
+      response =
+        request("/token.deploy_erc20", %{
+          symbol: "BTC",
+          name: nil,
+          amount: 1000,
+          locked: false,
+          description: "desc",
+          subunit_to_unit: 100
+        })
+
+      assert_deploy_erc20(:fails_with_missing_data, response)
+    end
+
+    test_with_auths "fails to deploys an ERC20 token with amount = nil", context do
+      enable_blockchain(context)
+
+      response =
+        request("/token.deploy_erc20", %{
+          symbol: "BTC",
+          name: "Bitcoin",
+          amount: nil,
+          locked: false,
+          description: "desc",
+          subunit_to_unit: 100
+        })
+
+      assert_deploy_erc20(:fails_with_missing_data, response)
+    end
+
+    test_with_auths "fails to deploys an ERC20 token with locked = nil", context do
+      enable_blockchain(context)
+
+      response =
+        request("/token.deploy_erc20", %{
+          symbol: "BTC",
+          name: "Bitcoin",
+          amount: 1000,
+          locked: nil,
+          description: "desc",
+          subunit_to_unit: 100
+        })
+
+      assert_deploy_erc20(:fails_with_missing_data, response)
+    end
+
+    test_with_auths "fails to deploys an ERC20 token with subunit_to_unit = nil", context do
+      enable_blockchain(context)
+
+      response =
+        request("/token.deploy_erc20", %{
+          symbol: "BTC",
+          name: "Bitcoin",
+          amount: 1000,
+          locked: false,
+          description: "desc",
+          subunit_to_unit: nil
+        })
+
+      assert_deploy_erc20(:fails_with_missing_data, response)
+    end
+
+    test_with_auths "fails to deploys an ERC20 token with amount < 0", context do
+      enable_blockchain(context)
+
+      response =
+        request("/token.deploy_erc20", %{
+          symbol: "BTC",
+          name: "Bitcoin",
+          amount: -1000,
+          locked: false,
+          description: "desc",
+          subunit_to_unit: 100
+        })
+
+      assert_deploy_erc20(:fails_with_amount_zero_or_less, response)
+    end
+
+    test_with_auths "fails to deploys an ERC20 token with subunit_to_unit < 0", context do
+      enable_blockchain(context)
+
+      response =
+        request("/token.deploy_erc20", %{
+          symbol: "BTC",
+          name: "Bitcoin",
+          amount: 1000,
+          locked: false,
+          description: "desc",
+          subunit_to_unit: -1
+        })
+
+      assert_deploy_erc20(:fails_with_subunit_to_unit_zero_or_less, response)
+    end
+
+    test_with_auths "fails to deploys an ERC20 token with amount = 0", context do
```

Comment:
Sorry my comment above about zero was about deployment not minting.

Reply:
@okalouti, you are right. We can deploy an erc20 contract with amount 0. I will adjust it.